src/main/java/com/yahoo/bullet/postaggregations/ComputationStrategy.java

```diff
@@ -0,0 +1,250 @@
+    private Computation postAggregation;
+
+    @Override
+    public Clip execute(Clip clip) {
+        clip.getRecords().forEach(r -> {
+                try {
+                    TypedObject result = calculate(postAggregation.getExpression(), r);
+                    switch (result.getType()) {
+                        case INTEGER:
+                            r.setInteger(postAggregation.getNewName(), (Integer) result.getValue());
+                            break;
+                        case LONG:
+                            r.setLong(postAggregation.getNewName(), (Long) result.getValue());
+                            break;
+                        case DOUBLE:
+                            r.setDouble(postAggregation.getNewName(), (Double) result.getValue());
+                            break;
+                        case FLOAT:
+                            r.setFloat(postAggregation.getNewName(), (Float) result.getValue());
+                            break;
+                        case BOOLEAN:
+                            r.setBoolean(postAggregation.getNewName(), (Boolean) result.getValue());
+                            break;
+                        case STRING:
+                            r.setString(postAggregation.getNewName(), (String) result.getValue());
+                            break;
+                    }
+                } catch (RuntimeException e) {
+                    // Ignore the exception and skip setting the field.
+                }
+            });
+        return clip;
+    }
+
+    @Override
+    public Optional<List<BulletError>> initialize() {
+        return Optional.empty();
+    }
+
+    private TypedObject calculateLeafExpression(LeafExpression expression, BulletRecord record) {
+        Value value = expression.getValue();
+        TypedObject result = null;
+        String valueString = value.getValue();
+        switch (value.getKind()) {
+            case FIELD:
+                result = extractTypedObject(valueString, record);
+                break;
+            case VALUE:
+                result = TypedObject.forceCast(value.getType(), valueString);
+                break;
+        }
+        return result;
+    }
+
+    private TypedObject calculateBinaryExpression(BinaryExpression expression, BulletRecord record) {
+        TypedObject left = calculate(expression.getLeft(), record);
+        TypedObject right = calculate(expression.getRight(), record);
+        Type resultType = getResultType(left.getType(), right.getType());
+        TypedObject castedLeft = left.forceCast(resultType);
+        TypedObject castedRight = right.forceCast(resultType);
+
+        TypedObject result = null;
+        switch (expression.getOperation()) {
+            case ADD:
+                result = add(resultType, castedLeft, castedRight);
+                break;
+            case SUB:
+                result = sub(resultType, castedLeft, castedRight);
+                break;
```

Comment:
Here you could just do `return sub(resultType, castedLeft, castedRight);` for all these and remove the "break" statements. Then return null at the end (and not bother initializing a "result" object).   Same is true for add() and sub().

I think I like that way better (with no unnecessary "break"s).  This way works too, but if you do it this way we should be consistent - Type. forceCast(), etc do it the other way with no breaks - so maybe change those if you want to do it this way?

Or change these to be consistent?

Reply:
In this way, you have to return null or throw unreachable exception at the end, otherwise Java compile will complain it. But it will reduce the code coverage since that line is not reachable.

Reply:
Gotcha.